state/metrics.go

```diff
@@ -161,6 +173,52 @@ func (st *State) AddMetrics(batch BatchParam) (*MetricBatch, error) {
 	return metric, nil
 }
 
+// AddModelMetrics adds a new model-centric batch of metrics to the database.
+func (st *State) AddModelMetrics(batch ModelBatchParam) (*MetricBatch, error) {
+	if len(batch.Metrics) == 0 {
+		return nil, errors.New("cannot add a batch of 0 metrics")
+	}
+	slaCreds, err := st.SLACredential()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	metric := &MetricBatch{
+		st: st,
+		doc: metricBatchDoc{
+			UUID:           batch.UUID,
+			ModelUUID:      st.ModelUUID(),
+			Sent:           false,
+			Created:        batch.Created,
+			Metrics:        batch.Metrics,
+			SLACredentials: slaCreds,
+		},
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
```

Comment:
do we really have to use transactions to insert metrics into a collection? seems a bit of an overkill...